### PR TITLE
GEOMESA-1137 Adding sampling in distributed iterators

### PR DESCRIPTION
**Pull request by elahrvivaz**
* Sampling can be triggered via query hints or a WPS process - geomesa:Sampling

Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AttributeIdxStrategy.scala [hunk 1]
```diff
@@ -72,14 +73,10 @@ class AttributeIdxStrategy(val filter: QueryFilter) extends Strategy with LazyLo
 
     // query against the attribute table
     val singleAttrValueOnlyPlan: ScanPlanFn = (schema, filter, transform) => {
-      val iterators = if (filter.isDefined || transform.isDefined) {
-        Seq(KryoLazyFilterTransformIterator.configure(schema, filter, transform, priority))
-      } else {
-        Seq.empty
-      }
+      val iters = KryoLazyFilterTransformIterator.configure(schema, filter, transform, sampling).toSeq
```
Comment (jahhulbert-ccri) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AttributeIdxStrategy.scala]: do we also need to update the AttrValuePlusKey itr?

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AttributeIdxStrategy.scala]: oh, let me check... that wasn't there when i first wrote this

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AttributeIdxStrategy.scala]: yeah, I need to add that, good catch

Comment (elahrvivaz) [inline geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/AttributeIdxStrategy.scala]: added.

**Comment by jahhulbert-ccri** [comment]
54af5042a9db90aca23f6e5c5ec1b4fa5ff1631c

**Pull request closed without merging**